### DISCUSSION TITLE
Jitter factor

### DISCUSSION THREAD

**Discussion by pandoras-toolbox** [junit-pioneer/junit-pioneer#778] [Ideas]
I do not know if you will implement it and need it now, so I copied your RetryingTest classes into my test automation project and added an option to specify a jitter factor for the delay.

Because if tests run in parallel and they are all retried at nearly the same time then the result is more likely to fail again, especially for E2E tests.

But if the start for retries varies randomly then it is a bit more likely that tests eventually succeed.

For instance if delay is set to 10 seconds and a jitter factor of 0.5 then the delay will randomly vary between 5 and 15 seconds.

What also might good to provide options to use a backoff strategy like in Failsafe for Java.

**Comment by beatngu13** [comment]
I like the idea of a [Jitter](https://en.wikipedia.org/wiki/Jitter) factor for retrying tests. What does the rest of the team think?

Regarding [your second comment](https://github.com/junit-pioneer/junit-pioneer/discussions/778#discussioncomment-7377835): would you mind opening a separate discussion/issue for it as this is somewhat orthogonal to the Jitter factor discussion here?

**Comment by pandoras-toolbox** [comment]
I deleted now the second topic here and will create a new discussion for that.

Regarding the Jitter factor, in the method adjustForJitter(...) you can see how it is [implemented](https://github.com/failsafe-lib/failsafe/blob/e8d99288a91df892340d325ad4562d064d405296/core/src/main/java/dev/failsafe/internal/RetryPolicyExecutor.java#L287C16-L287C26) in Failsafe.

**Comment by Bukama** [comment]
> I like the idea of a [Jitter](https://en.wikipedia.org/wiki/Jitter) factor for retrying tests. What does the rest of the team think?

I also like the idea. I will open an issue.

**Reply by beatngu13** [reply]
See: #783
